### PR TITLE
[bitnami/kafka]Use template for log4j configuration

### PR DESCRIPTION
**Pull request by johnpptz**
This will allow to use variables in the log4j configuration

log4j: |-
  log4j.rootLogger={{ default "INFO" .Values.logLevel }}, stdout, kafkaAppender
  log4j.appender.stdout=org.apache.log4j.ConsoleAppender
  log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
  log4j.appender.stdout.layout.ConversionPattern=[%d] %p %m (%c)%n

**Description of the change**

**Benefits**

**Possible drawbacks**

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/kafka/templates/log4j-configmap.yaml [hunk 1]
```diff
@@ -12,5 +12,5 @@ metadata:
   {{- end }}
 data:
   log4j.properties: |-
-    {{ .Values.log4j | nindent 4 }}
+{{ tpl .Values.log4j $ | indent 4 }}
```
Comment (marcosbc) [inline bitnami/kafka/templates/log4j-configmap.yaml:15]: I would keep the `nindent` + 4-space indentation for now, for consistency with how Bitnami charts are indented:

```suggestion
    {{ tpl .Values.log4j $ | nindent 4 }}
```

**Review by marcosbc — COMMENTED** [review]
Thanks for the contribution! I have a small request, please take a look at it whenever you have a chance.

File: bitnami/kafka/templates/log4j-configmap.yaml (hunk 1 above)
Comment (johnpptz) [reply bitnami/kafka/templates/log4j-configmap.yaml:15]: In this case we will get error.
/log4j-configmap.yaml: error converting YAML to JSON: yaml: line 13: did not find expected key

Comment (johnpptz) [reply bitnami/kafka/templates/log4j-configmap.yaml]: Thanks for good idea.

File: bitnami/kafka/templates/log4j-configmap.yaml [hunk 2]
```diff
@@ -12,5 +12,5 @@ metadata:
   {{- end }}
 data:
   log4j.properties: |-
-    {{ .Values.log4j | nindent 4 }}
+    {{ tpl .Values.log4j $ | nindent 4 }}
```
Comment (dani8art) [inline bitnami/kafka/templates/log4j-configmap.yaml:15]: Could we use our standard for tpl values as well, please?

```
{{- include "kafka.tplValue" ( dict "value" .Values.log4j "context" $ ) | nindent 4 }}
```

**Review by dani8art — CHANGES_REQUESTED** [review]
Hi thanks for applying the suggested changes, please, take a look at my comment.

File: bitnami/kafka/templates/log4j-configmap.yaml (hunk 2 above)
Comment (johnpptz) [reply bitnami/kafka/templates/log4j-configmap.yaml]: done

**Review by dani8art — APPROVED** [review]
Thanks! LGTM

**Pull request merged**